File: Python Track/ Week 3/ Day 2(Tue)/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # To store val of nums: with there index
        numMap = {}
        # To store the indexes
        result = []

        # to iterate over the hashmap and find the complement that brings the target and return there indexes else store the number in the hashmap
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in numMap:
                result.append(numMap[complement])
                result.append(i)
                break
            numMap[nums[i]] = i

        return result

File: Phase2/1-two-sum/two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var dict = Map<int,int>();
    List<int> result = [];

    for(var i = 0;i < nums.length;i++){
        var complement = target - nums[i];
        if(dict.containsKey(complement)){
            result.add(dict[complement]!);
            result.add(i);
            break;
        }
        dict[nums[i]] = i;

    } 
    return result;
  }
}